dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/FdcService.java

```diff
@@ -168,123 +181,118 @@ private Integer updateFdcAndCreateFile(List<Fdc> successfulFdcs, Map<String, Str
-            log.error("FDC global update threw an exception [" + e.getClass().getName() + "(" + e.getMessage() + ")]");
-            throw e;
         }
     }
 
-    private int callGlobalUpdate() {
-        FdcGlobalUpdateResponse globalUpdateResponse;
-        try {
-            globalUpdateResponse = callFdcGlobalUpdate();
-        } catch (Exception e) {
-            return logGlobalUpdateFailure(e.getMessage());
-        }
-        // handle response
-        if (!globalUpdateResponse.isSuccessful()) {
-            return logGlobalUpdateFailure("endpoint response has successful==false");
-        }
-        int numUpdates = globalUpdateResponse.getNumberOfUpdates();
-        eventService.logFdc(FDC_GLOBAL_UPDATE, batchId, null, OK, "Updated "+numUpdates+" fdc entries");
-        return numUpdates;
-    }
-
-    private int logGlobalUpdateFailure(String errorMessage){
-        // We've failed to do a global update.
-        // TODO: Flag here for sentry
-        String errorText = String.format("Failed to complete FDC global update [%s]", errorMessage);
-        log.error(errorText);
-        eventService.logFdc(FDC_GLOBAL_UPDATE, batchId, null, INTERNAL_SERVER_ERROR, errorText);
-        // continue processing, as can still have data to deal with.
-        return 0;
-    }
-
     @Retry(name = SERVICE_NAME)
-    public Integer fdcUpdateRequest(String xmlContent, List<String> fdcIdList, int numberOfRecords, String fileName, String fileAckXML) throws HttpServerErrorException {
+    private Integer executeFdcUpdateRequestCall(String xmlContent, List<String> fdcIdList, int numberOfRecords, String fileName, String fileAckXML) {
         FdcUpdateRequest request = FdcUpdateRequest.builder()
                 .recordsSent(numberOfRecords)
                 .xmlContent(xmlContent)
                 .fdcIds(fdcIdList)
                 .xmlFileName(fileName)
                 .ackXmlContent(fileAckXML).build();
         if (!feature.outgoingIsolated()) {
-            return fdcClient.updateFdcs(request);
+            try {
+                return fdcClient.updateFdcs(request);
+            } catch (WebClientResponseException e){
+                logFileCreationError(e, e.getStatusCode());
+                throw e;
+            }
         } else {
             log.info("fdcUpdateRequest: Not calling MAAT API updateFdcs() because `feature.outgoing-isolated=true`");
             return 0;
         }
     }
 
-    private void logMaatUpdateEvents(List<Fdc> successfulFdcs, Map<String, String> failedFdcs) {
+    // Logging Methods
+
+    private void logNumberDiscepancies(int globalUpdateCount, int getFdcCount, int successfullySentFdcCount) {
+        if (globalUpdateCount != getFdcCount) {
+            log.info("FDC contribution count discrepancy: {} affected by FDC global update, {} from getFdcs", globalUpdateCount, getFdcCount);
```

Comment:
Should these discrepancies be logged in the events database also, so that all observations are there for easy future access if needed?

Reply:
These were an old addition that I moved about. I've spoken with Tariq, and we've both agreed that the new DB logging will show all this detail already. 
FDC Global Update will log how many affected, Get how many fetched, and the sendToDRC how many it sent.